packages/source-iotsitewise/src/time-series-data/subscribeToTimeSeriesData.ts

```diff
@@ -36,41 +44,21 @@ export const subscribeToTimeSeriesData =
       emit();
     });
 
-    const fetchResources = ({ queries }: { queries?: SiteWiseDataStreamQuery[] }) => {
-      if (queries) {
-        queries.forEach((query) => {
-          query.assets.forEach((asset) => {
-            assetModuleSession
-              .fetchAssetSummary({ assetId: asset.assetId })
-              .then((assetSummary) => {
-                if (assetSummary && assetSummary.assetModelId != null) {
-                  return assetModuleSession.fetchAssetModel({ assetModelId: assetSummary.assetModelId });
-                }
-              })
-              .then((assetModelResponse) => {
-                if (assetModelResponse) {
-                  assetModels[asset.assetId] = assetModelResponse;
-                  emit();
-                }
-              })
-              .catch((err: ErrorDetails) => {
-                // TODO: Currently these are not used anywhere. Do something with these errors.
-                errors[asset.assetId] = err;
-                // emit();
-              });
-          });
-        });
-      }
-    };
-    fetchResources({ queries });
+    emitAssetModels({ emit, queries, assetModuleSession, errors, assetModels });
```

Comment:
This really isn't "emit"ing anything. Maybe just fetchAssetModels, fetchAlarms?

Reply:
it is calling emit, it is emiting.

Reply:
I'm more referring to semantics where "emit" implies there is an "emitter" that emits more than once, in response to some event. I guess there's a world where we `subscribeToAlarms` or `subscribeToAssetData` in which "emit" makes sense here.